src/core/EventService.js

```diff
@@ -0,0 +1,46 @@
+export default class EventService {
+  constructor(socket) {
+    this._socket = socket;
+    this._refs = {};
+  }
+
+  _registerEvent(name, label, event, cb) {
+    const ref = this._socket.channel(name).on(event, cb);
+    this._refs[label + ':' + event] = ref;
+  }
+
+  _unregisterEvent(name, label, event) {
+    const ref = this._refs[label + ':' + event];
+    delete this._refs[label + ':' + event];
+    this._socket.channel(name).off(event, ref);
+  }
+
+  _cleanEventName(event) {
+    switch (event) {
+      case 'start':
+        return 'started';
+      case 'stop':
+        return 'stopped';
+      case 'start_existing':
+        return 'started';
+      case 'take_snapshot':
+        return 'snapshot_taken';
+      case 'revert_snapshot':
+        return 'snapshot_reverted';
+      default:
+        return event;
+    }
+  }
+
+  once(channel, event, cb) {
```

Comment:
i think there's no need for an EventService class if `once` is the main thing you want to achieve.

```js
export function listenOnce(socket, channel, event, cb) {
  checkValidEventName(event);
  const handler = socket.channel(channel).on(event, data => {
    socket.channel(channel).off(event, handler);
    cb(data);
  })
}
```

Reply:
I haven't included them yet but there are a number of default events for each chain which are to be used. 

This is a bit of a grey area at the minute as outlined in the wireframes (I've emailed them to you) there is a terminal-like display for deployment of chains. If we are to direct logging based on events I would maybe want to seperate that functionality to this class